d2bs/kolbot/libs/common/Pather.js

```diff
@@ -617,6 +617,97 @@ ModeLoop:
 					/* i < areas.length - 1 is for crossing multiple areas.
 						In that case we must use the exit before the last area.
 					*/
+					if (targetArea == 65) {
+						unit = getUnit(2, 74);
+						if (unit) {
+							switch (unit.mode) {
+							case 0:
+								sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
+								delay(500);
+								break;
+							case 2:
+								break;
+							}
+							sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
+							return true;
+						} else {
+							throw new Error("moveToExit: Unit not found.");
+						}
+					}
+					if (me.area == 92 && targetArea == 93) {
+						this.moveToPreset(92, 2, 366);
+						lever = getUnit(2, 367);
+						unit = getUnit(2, 366);
+						if (unit) {
+							switch (unit.mode) {
+							case 0:
+								if (lever) {
+									sendPacket(1, 0x13, 4, 0x02, 4, lever.gid);
+									delay(500);
+									break;
+								} else {
+									throw new Error("moveToExit: Lever unit not found.");
+								}
+							case 2:
+								break;
+							}
+							sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
+							return true;
+						} else {
+							throw new Error("moveToExit: Unit not found.");
+						}
+					}
+					if (targetArea == 94 || targetArea == 95 || targetArea == 96 || targetArea == 97 || targetArea == 98 || targetArea == 99) {
+						unit = getUnit(2, "stair");
+						if (unit) {
+							switch (unit.mode) {
+							case 0:
+								sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
+								delay(500);
+								break;
+							case 2:
+								break;
+							}
+							sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
+							return true;
+						} else {
+							throw new Error("moveToExit: Unit not found.");
+						}
+					}
+					if (me.area == 120 && targetArea == 128) {
+						unit = getUnit(2, 547);
+						if (unit) {
+							switch (unit.mode) {
+							case 0:
+								if (me.getQuest(39, 0) == 1) {
+									sendPacket(1, 0x13, 4, 0x02, 4, unit.gid);
```

Comment:
Can you explain what is going on here?
1. There is a switch case for 2 items, unit.mode 0 and 2, but in case 2 it does nothing. I don't think a switch case is needed here.

2. In case 0 it sends a packet, breaks the switch case, then sends the same packet again in line 692?

This same pattern occurs multiple times.

Reply:
Msg me on IRC if you get a chance.